### PR TITLE
PUD-766: Endpoint to return summary of audit information for recall

### PR DESCRIPTION
**Pull request by philsegal**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/component/RecallAuditComponentTest.kt [hunk 1]
```diff
@@ -2,65 +2,87 @@ package uk.gov.justice.digital.hmpps.managerecallsapi.component
...
 
-  @Autowired private lateinit var recallAuditService: RecallAuditService
-
   @Test
   fun `get currentPrison audit for a recall that has been booked`() {
-    val recallId = ::RecallId.random()
-    val createdByUserId = authenticatedClient.userId
-
-    val now = OffsetDateTime.ofInstant(Instant.parse("2021-10-04T14:15:43.682078Z"), ZoneId.of("UTC"))
+    val savedRecall = authenticatedClient.bookRecall(BookRecallRequest(nomsNumber, FirstName("Brian"), null, LastName("Badgering")))
```
Comment (AndyNewton) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/component/RecallAuditComponentTest.kt:26]: Might be clearer if this person name is e.g. "The Offender".  Or if we switch from "Bertie Badger" to "Current User"?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/component/RecallAuditComponentTest.kt [hunk 2]
```diff
@@ -2,65 +2,87 @@ package uk.gov.justice.digital.hmpps.managerecallsapi.component
...
-      Recall(recallId, nomsNumber, createdByUserId, now, FirstName("Brian"), null, LastName("Badgering"), currentPrison = PrisonId("ABC")),
-      createdByUserId
-    )
+    val updatedRecall = authenticatedClient.updateRecall(recallId, UpdateRecallRequest(currentPrison = PrisonId("ABC")))
 
     val auditList = authenticatedClient.auditForField(recallId, "currentPrison")
 
     assertThat(auditList.size, equalTo(1))
     assertThat(auditList[0].updatedValue, equalTo("ABC"))
     assertThat(auditList[0].updatedByUserName, equalTo(FullName("Bertie Badger")))
```
Comment (AndyNewton) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/component/RecallAuditComponentTest.kt:35]: QQ: Where do we set "Bertie Badger"?

**Review by AndyNewton — APPROVED** [review]
Sorry for delay; looks good; only minor QQs.
ps.  Where has the API been spec'ed please?  Ta.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/db/Converters.kt [hunk 3]
```diff
@@ -12,6 +13,7 @@ import uk.gov.justice.digital.hmpps.managerecallsapi.domain.PrisonId
 import javax.persistence.AttributeConverter
 
 class FirstNameJpaConverter : CustomJpaConverter<FirstName, String>({ it.value }, ::FirstName)
+class FullNameJpaConverter : CustomJpaConverter<FullName, String>({ it.value }, ::FullName)
```
[the remark this answers is not in the document]
Comment (philsegal) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/db/Converters.kt:16]: Good spot

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/component/RecallAuditComponentTest.kt (hunk 1 above)
Comment (philsegal) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/component/RecallAuditComponentTest.kt:26]: Yep, probably a good change at some point. Cassie Caseworker and Ollie Offender?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/component/RecallAuditComponentTest.kt (hunk 2 above)
Comment (philsegal) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/component/RecallAuditComponentTest.kt:35]: In `@BeforeAll` of `ComponentTestBase`. Line 110.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/controller/AuditController.kt [hunk 4]
```diff
@@ -35,3 +41,11 @@ data class FieldAuditEntry(
   val updatedDateTime: OffsetDateTime,
   val updatedValue: Any
 )
+
+data class FieldAuditSummary(
+  val auditId: Int,
+  val fieldName: FieldName,
```
[the remark this answers is not in the document]
Comment (philsegal) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/controller/AuditController.kt:47]: This object doesnt care about type. But for the next story we probably need to find a way to model it.

**Pull request merged**